database/index.js

```diff
@@ -66,8 +66,16 @@ User.findById(data.project_id, function (err, user) {
 			  project.save();
            });
 }
-var deleteProject = function(data,callback){
-	//console.log("Dataaaa",data)
+var deleteProject = function(data,userId,callback){
+	User.findById(userId,function(err,user){
+		if(err){throw err}
+			for(var i=0; i<user.projects.length;i++){
+				if(user.projects[i]._id.toString() === data._id){
```

Comment:
You're going to want to account for a situation where the user isn't found. If `user` is null, `user.projects` is going to throw an error. To avoid this, throw in an if statement that checks to make sure `user` is defined, and, if it isn't, sends a 404 response.